### PR TITLE
[WebDiscover] Determine if IAM policy setup step can be skipped

### PR DESCRIPTION
**Pull request by kimlisa**
depends on https://github.com/gravitational/teleport/pull/29721

Checks if the registered database has a configured IAM policy or not (it's a screen where you create and attach an policy to the ec2 instance you deployed the db service in). Depending on this state, a user can skip both the deploy service screen AND the IAM policy screen.

Relevant screenshots

make it apparent user can skip this screen:
image

if user encounters error with auto deploy, render a note that it could be b/c changes has not finished taking affect:
image

### CODE REVIEW

**Comment by ibeckermayer** [comment]
Should I wait for https://github.com/gravitational/teleport/pull/29721 to land to review this?

**Comment by kimlisa** [comment]
> Should I wait for #29721 to land to review this?

no, the only thing that will probably change is just the `enum` value might be a string value now. but underlying logic won't change

**Comment by kimlisa** [comment]
@ibeckermayer @ryanclark ready for review

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx:372]: The comment says polling should continue, but these lines don't check that anything has been called twice. What am I missing?

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx:399]: If it's skipping two steps then why has it been called three times?

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts:144]: Is there any case where we have `!isAws === true` but also that `dbServer.aws` is defined?

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts:150]: This logic is a bit difficult for me to follow, consider if you can simplify and/or make the comment more clear.

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx:204]: How do they "attempt again" here? Does the `Deploy Teleport Service` button need to be re-enabled? It's not clear, but it seems as though it may be disabled in your screenshot.

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx:372]: i advance the timer again on line 386 and not calling `updateAgentMeta` means it hasn't stopped polling (we only call it after a result came back)

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.test.tsx:399]: this is how `nextStep(numToIncrement)` interprets numbers:

0 = user intentionally skip the current step
undefined, null, 1 = user finished with the current step
greater than 1 = we skip for the user

so a skip number 3 means:
  1 = finished current step
 +1 = skip deploy step
 +1 = skip iam policy step

File: web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx:204]: by clicking on the "deploy teleport service" button

non error state: 
image

error state:
image

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts
Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts:144]: no, just wanted to be safe before accessing a possible undefined field for whatever reason

**Comment by kimlisa** [comment]
friendly ping @ibeckermayer @ryanclark

**Review by r0mant — APPROVED** [review]
Approving just so you're not blocked on waiting for 2nd approval as we need to ship this.

File: web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts
Comment (ibeckermayer) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts:153]: I believe these are equivalent:
```suggestion
          const dbServer = res.agents[0];
          if (
            !isAws || // If not AWS, then we return the first thing we get back.
            dbServer.aws?.iamPolicyStatus !== IamPolicyStatus.Pending // If AWS and aws.iamPolicyStatus is undefined or non-pending, we return the dbServer.
          ) {
            return dbServer;
          }
          // Otherwise continue polling
```

Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/useCreateDatabase.ts]: done

**Review by ryanclark — APPROVED** [review]

**Pull request merged**
